tests/unit/src/test/scala/tests/testProvider/DiscoverTestClassesSuite.scala

```diff
@@ -0,0 +1,109 @@
+package tests
+
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.ServerCommands
+import scala.meta.internal.metals.testProvider.TestDiscovery
+import scala.meta.internal.metals.testProvider.TestDiscoveryResult
+
+class DiscoverTestClassesSuite extends BaseLspSuite("discoverTests") {
+
+  test("test-classes") {
+    for {
+      _ <- Future.successful(cleanWorkspace())
+      _ <- initialize(s"""|/metals.json
+                          |{
+                          |  "app": {
+                          |    "libraryDependencies" : [ "org.scalameta::munit:0.7.29", "org.scalatest::scalatest:3.2.1" ],
+                          |    "scalaVersion": "2.13.6"
+                          |  }
+                          |}
+                          |/app/src/main/scala/foo/bar/MyTestSuite.scala
+                          |package foo.bar
+                          |abstract class AbstractTestSuite1 extends munit.FunSuite 
+                          |
+                          |class MunitTestSuite extends AbstractTestSuite1 {
+                          |  test("foo") {}
+                          |}
+                          |""".stripMargin)
+      _ <- server.server.compilations.compilationFinished(
+        workspace.resolve("app/src/main/scala/foo/bar/MyTestSuite.scala")
+      )
+      _ <- server.executeCommand(ServerCommands.CascadeCompile)
+      _ <- server.didOpen("app/src/main/scala/foo/bar/MyTestSuite.scala")
+      _ <- server.didSave("app/src/main/scala/foo/bar/MyTestSuite.scala")(
+        identity
+      )
+      res <- server.testClasses("app/src/main/scala/foo/bar/MyTestSuite.scala")
+    } yield {
+      val obtained = res.map(TestDiscoveryWrapper(_))
+      val expected = List(
+        TestDiscoveryWrapper(
+          "app",
+          List(
+            TestDiscoveryResultWrapper.Package(
+              "foo",
+              List(
+                TestDiscoveryResultWrapper.Package(
+                  "bar",
+                  List(
+                    TestDiscoveryResultWrapper.TestClass(
+                      "foo.bar.MunitTestSuite",
+                      "MunitTestSuite"
+                    )
+                  )
+                )
+              )
+            )
+          )
+        )
+      )
+
+      assertEquals(expected, obtained)
+    }
+  }
+}
+
+final case class TestDiscoveryWrapper(
+    targetName: String,
+    discovered: List[TestDiscoveryResultWrapper]
+)
+object TestDiscoveryWrapper {
+  def apply(discovery: TestDiscovery): TestDiscoveryWrapper =
+    TestDiscoveryWrapper(
+      discovery.targetName,
+      discovery.discovered.asScala
+        .map(x => TestDiscoveryResultWrapper(x))
+        .toList
+    )
+}
+
+sealed trait TestDiscoveryResultWrapper
```

Comment:
It's quite a lot of boilerplate, but asserting logic is easier. Maybe someone has a better proposition